lib/url-updater/update-url-for-results.js

```diff
@@ -1,27 +1,43 @@
 'use strict';
 
+const RepoDataClient = require('@financial-times/origami-repo-data-client');
+const semver = require('semver');
+
 /**
  * @param {URL} url - The original Build Service URL
  * @param {ModuleResult[]} results - Details about the requested modules and how up to date they are.
  * @return {URL} - A Build Service URL updated for the latest releases of components.
  */
-module.exports = function updateUrlForResults(url, results) {
+module.exports = async function updateUrlForResults(url, results) {
 	const hasOnlySpecV1Components = results.every(v => v.latestOrigamiSpec === '1');
-	const minimumBuildServiceVersion = hasOnlySpecV1Components ? 2 : 3;
+	const hasComponentsBehindTheLatestBowerMajor = results
+		.some(component => component.belowLatestBowerRelease);
+
+	const minimumBuildServiceVersion = hasOnlySpecV1Components || hasComponentsBehindTheLatestBowerMajor ? 2 : 3;
 
 	const updatedUrl = new URL(
 		url.toString().replace(
 			/(?<=service\/build\/v)([1-2])(?=\/)/,
 			minimumBuildServiceVersion
 		));
 
-
 	const componentParam = minimumBuildServiceVersion === 2 ? 'modules' : 'components';
 	const unsupportedComponentParam = minimumBuildServiceVersion === 2 ? 'components' : 'modules';
+
+	const componentParamValue = results.map(c => `${c.name}@${hasComponentsBehindTheLatestBowerMajor ? (c.belowLatestBowerRelease ? `^${c.lastBowerMajor}` : c.requestedVersion) : `^${c.latestVersion}`}`);
```

Comment:
non-blocking suggestion: I had a hard time reading the nested template literals and ternaries here. To understand what was going on, I changed it to a nested if statement for setting only the version part of the final string.
Does this look correct @notlee?
```suggestion
	const componentParamValue = results.map(c => {
		let version = `^${c.latestVersion}`;
		if (hasComponentsBehindTheLatestBowerMajor) {
			if (c.belowLatestBowerRelease) {
				version = `^${c.lastBowerMajor}`;
			} else {
				version = c.requestedVersion;
			}
		}
		return `${c.name}@${version}`;
	};
```